create_index.py

```diff
@@ -0,0 +1,116 @@
+def format_as_tuple(node_name, properties):
+    """
+    Format index info as a tuple
+    :param node_name: The name of the node type for the index
+    :param properties: The list of node properties being used by the index
+    :return: A tuple containing the index node_name followed by the index properties in alphabetical order
+    """
+    if isinstance(properties, str):
+        properties = [properties]
+    lst = [node_name] + sorted(properties)
+    return tuple(lst)
+
+def create_index(driver, schema, log, database_type):
+    index_created = 0
+    if database_type == "neo4j":
+        with driver.session() as session:
+            tx = session.begin_transaction()
+            try:
+                index_created = create_neo4j_indexes(tx, schema, log)
+                tx.commit()
+            except Exception as e:
+                tx.rollback()
+                log.exception(e)
+                return False
+    elif database_type == "memgraph":
+        try:
+            cursor = driver.cursor()
+            index_created = create_memgraph_indexes(cursor, schema, log)
+            #tx.commit()
+            #self.mg_connection.commit()
+        except Exception as e:
+            #tx.rollback()
+            log.exception(e)
+            return False
+    return index_created
+
+def get_btree_indexes(session):
+    """
+    Queries the database to get all existing indexes
+    :param session: the current neo4j transaction session
+    :return: A set of tuples representing all existing indexes in the database
+    """
+    command = "SHOW INDEXES"
+    result = session.run(command)
+    indexes = set()
+    for r in result:
+        if r["type"] == "BTREE":
+            indexes.add(format_as_tuple(r["labelsOrTypes"][0], r["properties"]))
+    return indexes
+
+def create_neo4j_indexes(session, schema, log):
```

Comment:
This function has almost identical code as create_memgraph_indexes, and should be combined into one if possible.